### PR TITLE
Add support for dirty rects

### PR DESCRIPTION
**Pull request by pioto1225**
Hi,
I would like to add support for dirty rects using atomic mode setting. Corresponding Gnome/Wayland [change](https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1879) is under review. This was tested on Fedora 34, and it works pretty neat. I am looking forward to your feedback.
Thanks.
Piotr

PS. Just for reference: Ubuntu/Debian prototype using legacy KMS [see here](https://github.com/pioto1225/evdi/tree/dirty_rects)

### CODE REVIEW

**Review by lspintzyk — COMMENTED** [review]
1. On which distro/desktop environment/xorg/wayland it is tested? 
2. This needs to build as well for older kernels then 5.0
Please use ./ci/build_against_kernel to test it.

**Comment by pioto1225** [comment]
1. Partial updates were tested on Fedora 34 Gnome/Wayland (including Mutter changes). Mutter changes are for Gnome/Wayland only, other compositors could add support. I tested EVDI changes for regression on stock Fedora 34, Debian Bullseye (Gnome Wayland), Ubuntu 20.04 (Gnome Wayland and X11). All OK. 

2. Fixed now, thanks.

File: module/evdi_modeset.c [hunk 1]
```diff
@@ -251,10 +260,33 @@ static void evdi_plane_atomic_update(struct drm_plane *plane,
...
+			state->src.y2 = fb->height << 16;
+
+			drm_atomic_helper_damage_iter_init(&iter, old_state, state);
+			while (drm_atomic_helper_damage_iter_next(&iter, &rect)) {
+				clip_rect.x1 = rect.x1;
+				clip_rect.y1 = rect.y1;
+				clip_rect.x2 = rect.x2;
+				clip_rect.y2 = rect.y2;
+				evdi_painter_mark_dirty(evdi, &clip_rect);
+				EVDI_DEBUG("Atomic update, dirty rect [%d, %d, %d, %d]\n",
```
Comment (lspintzyk) [inline module/evdi_modeset.c:278]: I think that evdi_painter_mark_dirty is already logging in dirty rects, and it is doing that in VERBOSE mode. Ihom this line is to be deleted.
Also default log mode is DEBUG so this log is to verbose.

**Review by lspintzyk — APPROVED** [review]
Thanks for doing this! This is fine to me.
Approving and we will merge it soon :)

File: module/evdi_modeset.c [hunk 2]
```diff
@@ -26,6 +26,9 @@
 #include "evdi_cursor.h"
 #include "evdi_params.h"
 #include <drm/drm_gem_framebuffer_helper.h>
+#if KERNEL_VERSION(5, 0, 0) <= LINUX_VERSION_CODE
```
Comment (elguero) [inline module/evdi_modeset.c:29]: It looks like EL8 has this support present.  You should add a check for it:

`#if KERNEL_VERSION(5, 0, 0) <= LINUX_VERSION_CODE || defined(EL8)`

**Review by elguero — CHANGES_REQUESTED** [review]

**Comment by pioto1225** [comment]
Great! I have added two changes that were requested.
Thanks!

**Comment by displaylink-emajewsk** [comment]
Merged. Thanks again, @pioto1225 ! 😇

**Comment by pioto1225** [comment]
thank you!

**Pull request merged**